### PR TITLE
Add storefact

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Review by fjetter — APPROVED** [review]

File: docs/index.rst [hunk 1]
```diff
@@ -46,6 +46,50 @@ Note that by changing the first two lines to::
...
+
+    from minimalkv import get_store_from_url, get_store
+
+    store = get_store_from_url('azure://test:XXXsome_azure_account_keyXXX@my-azure-container')
+    store.put(u'key', b'value')
+    assert store.get(u'key') == b'value'
+
+URL and store types:
+
+* In memory: :code:`memory://` and :code:`hmemory://`.
```
Comment (mlondschien) [inline docs/index.rst:81]: Maybe add examples here?

File: minimalkv/__init__.py [hunk 2]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
         self._copy(source, dest)
         self._delete(source)
         return dest
+
+
+def get_store_from_url(url):
+    """
+    Take a URL and return a simplekv store according to the parameters in the URL.
```
Comment (mlondschien) [inline minimalkv/__init__.py:515]:
```suggestion
    Take a URL and return a minimalkv store according to the parameters in the URL.
```

File: minimalkv/__init__.py [hunk 3]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
...
+    return get_store(**url2dict(url))
+
+
+def get_store(type, create_if_missing=True, **params):
+    """Return a storage object according to the `type` and additional parameters.
+
+    The *type* must be one of the types below, where each allows
+    different parameters:
+
+    * ``"azure"``: Returns a ``simplekv.azure.AzureBlockBlobStorage``. Parameters are
```
Comment (mlondschien) [inline minimalkv/__init__.py:551]:
```suggestion
    * ``"azure"``: Returns a ``minimalkv.azure.AzureBlockBlobStorage``. Parameters are
```

File: minimalkv/__init__.py [hunk 4]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
...
+      ``"account_name"``, ``"account_key"``, ``"container"``, ``"use_sas"`` and ``"create_if_missing"`` (default: ``True``).
+      ``"create_if_missing"`` has to be ``False`` if ``"use_sas"`` is set. When ``"use_sas"`` is set,
+      ``"account_key"`` is interpreted as Shared Access Signature (SAS) token.FIRE
+      ``"max_connections"``: Maximum number of network connections used by one store (default: ``2``).
+      ``"socket_timeout"``: maximum timeout value in seconds (socket_timeout: ``200``).
+      ``"max_single_put_size"``: max_single_put_size is the largest size upload supported in a single put call.
+      ``"max_block_size"``: maximum block size is maximum size of the blocks(maximum size is <= 100MB)
+    * ``"s3"``: Returns a plain ``simplekv.net.botostore.BotoStore``.
```
Comment (mlondschien) [inline minimalkv/__init__.py:559]:
```suggestion
    * ``"s3"``: Returns a plain ``minimalkv.net.botostore.BotoStore``.
```

File: minimalkv/__init__.py [hunk 5]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
...
+      Parameters must include ``"host"``, ``"bucket"``, ``"access_key"``, ``"secret_key"``.
+      Optional parameters are
+
+       - ``"force_bucket_suffix"`` (default: ``True``). If set, it is ensured that
+         the bucket name ends with ``-<access_key>``
+         by appending this string if necessary;
+         If ``False``, the bucket name is used as-is.
+       - ``"create_if_missing"`` (default: ``True`` ). If set, creates the bucket if it does not exist;
+         otherwise, try to retrieve the bucket and fail with an ``IOError``.
+    * ``"hs3"`` returns a variant of ``simplekv.net.botostore.BotoStore`` that allows "/" in the key name.
```
Comment (mlondschien) [inline minimalkv/__init__.py:569]:
```suggestion
    * ``"hs3"`` returns a variant of ``minimalkv.net.botostore.BotoStore`` that allows "/" in the key name.
```

File: minimalkv/__init__.py [hunk 6]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
...
+      The parameters are the same as for ``"s3"``
+    * ``"fs"``: Returns a ``simplekv.fs.FilesystemStore``. Specify the base path as "path" parameter.
```
Comment (mlondschien) [inline minimalkv/__init__.py:571]:
```suggestion
    * ``"fs"``: Returns a ``minimalkv.fs.FilesystemStore``. Specify the base path as "path" parameter.
```

File: minimalkv/__init__.py [hunk 7]
```diff
@@ -503,3 +508,104 @@ def _move(self, source, dest):
...
+    * ``"hfs"`` returns a variant of ``simplekv.fs.FilesystemStore``  that allows "/" in the key name.
```
Comment (mlondschien) [inline minimalkv/__init__.py:572]:
```suggestion
    * ``"hfs"`` returns a variant of ``minimalkv.fs.FilesystemStore``  that allows "/" in the key name.
```

**Review by mlondschien — COMMENTED** [review]
We should also align imports to only use absolute imports at some point.

File: docs/index.rst (hunk 1 above)
Comment (xhochy) [reply docs/index.rst:81]: These are the explicit examples, the full URL is just `memory://` with no further paths.

**Comment by xhochy** [comment]
CI is failing because of https://www.githubstatus.com/incidents/cj7gzzj30411

**Comment by xhochy** [comment]
CI is green again 💚

File: docs/index.rst [hunk 8]
```diff
@@ -46,6 +46,50 @@ Note that by changing the first two lines to::
...
+* Redis: :code:`redis://[[password@]host[:port]][/db]` and :code:`hredis://[[password@]host[:port]][/db]`
+* Filesystem: :code:`fs://` and :code:`hfs://`
```
Comment (mlondschien) [inline docs/index.rst:83]:
```suggestion
* Filesystem: :code:`fs://` and :code:`hfs://`, e.g., :code:`hfs:///home/user/data`
```

File: docs/index.rst [hunk 9]
```diff
@@ -46,6 +46,50 @@ Note that by changing the first two lines to::
...
+* Amazon S3: :code:`s3://access_key:secret_key@endpoint/bucket[?create_if_missing=true]` and :code:`hs3://access_key:secret_key@endpoint/bucket[?create_if_missing=true]`
+* Azure Blob Storage (:code:`azure://` and :code:`hazure://`):
+    * with storage account key: :code:`azure://account_name:account_key@container[?create_if_missing=true][?max_connections=2]`
```
Comment (mlondschien) [inline docs/index.rst:86]:
```suggestion
    * with storage account key: :code:`azure://account_name:account_key@container[?create_if_missing=true][?max_connections=2]`, e.g., :code:`azure://MYACCOUNT:passw0rd!@bucket_66?create_if_missing=true`
```

**Review by mlondschien — COMMENTED** [review]
Not sure what `shared_access_signatures` would look like. Also I am not sure what a realistic AWS url would be.

**Review by mlondschien — APPROVED** [review]
No strong opinions though.

**Pull request merged**